### ISSUE TITLE
SPIKE: Improve modals using <dialog>

### ISSUE THREAD

**Issue by orion-cengage** [cengage/react-magma#1342]
CENG_3417

[https://react-magma.cengage.com/version/4.4.0/api/modal/](https://react-magma.cengage.com/version/4.4.0/api/modal/)

### Description:
Since our last testing of this component, the native `<dialog>` element now has greater browser support and should be considered for all modal dialogs. If your browser support requirements can accommodate the use of `<dialog>`, several keyboard and focus management considerations are automatically handled.

In particular, `<dialog>` it provides a method `showModal()` to display the dialog and a `close()` method to close it. When shown modally, it prevents interactions with the rest of the page without additional markup. Using `<dialog>` also provides a ::backdrop CSS pseudo element for easy styling of the overlay beneath a modal dialog.

See [https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) for more information.

### Acceptance Criteria

- What is the level of effort to upgrade our Modals to use `<dialog>`?
- Would it be a breaking change? How does it affect our adopters? 
- Document all findings and present to team
- If this is feasible and we want to move forward, create a new ticket to do the work
